libraries/AP_SerialManager/AP_SerialManager.cpp

```diff
@@ -646,6 +672,31 @@ AP_HAL::UARTDriver *AP_SerialManager::get_serial_by_id(uint8_t id)
     if (id < SERIALMANAGER_NUM_PORTS) {
         return hal.serial(id);
     }
+#if AP_SERIALMANAGER_REGISTER_ENABLED
+    for (auto p = registered_ports; p; p = p->next) {
```

Comment:
we should wrap this in 
`if (id >= AP_SERIALMANAGER_NET_PORT_1 && id <= AP_SERIALMANAGER_NET_PORT_1 + AP_NETWORKING_NUM_PORTS)`